workflow/controller/controller.go

```diff
@@ -1270,69 +1239,61 @@ func (wfc *WorkflowController) newPodInformer(ctx context.Context) (cache.Shared
 	return informer, nil
 }
 
-func (wfc *WorkflowController) newConfigMapInformer() (cache.SharedIndexInformer, error) {
-	indexInformer := v1.NewFilteredConfigMapInformer(wfc.kubeclientset, wfc.GetManagedNamespace(), 20*time.Minute, cache.Indexers{
+var watchControllerSemaphoreConfigMaps = os.Getenv("WATCH_CONTROLLER_SEMAPHORE_CONFIGMAPS") != "false"
+
+func (wfc *WorkflowController) newConfigMapInformerManaged() (cache.SharedIndexInformer, error) {
+	indexInformer := v1.NewConfigMapInformer(wfc.kubeclientset, wfc.GetManagedNamespace(), 20*time.Minute, cache.Indexers{
 		indexes.ConfigMapLabelsIndex: indexes.ConfigMapIndexFunc,
```

Comment:
just noting here that this indexer [is used](https://github.com/argoproj/argo-workflows/blob/934bf179a703a365a6a5eddfa6edb788492eac79/workflow/controller/cache_gc.go#L28) for more than just plugins. that's why it previously only had a label selector on the label and did not specify the plugin type (and then why it filtered in the `FilterFunc` by plugin type instead -- it would populate the indexer but not the cache)